DentOS_Framework/DentOsTestbed/src/dent_os_testbed/test/test_suite/functional/igmp/test_igmp_snooping.py

```diff
@@ -0,0 +1,612 @@
+                      2: {"igmp_msg_v": "igmpv2", "igmp_type": "22", "totalLength": "28"},
+                      3: {"igmp_msg_v": "igmpv3MembershipReport", "igmp_type": "22", "totalLength": "40"}}
+
+    # 1.Create bridge and enable IGMP Snooping, enslave all TG ports to bridge interface
+    # Config Fastleave on 1st rx_port
+    await common_bridge_and_igmp_setup(dev_name, bridge, igmp_ver, dut_ports)
+    out = await BridgeLink.set(
+        input_data=[{dev_name: [
+            {"device": dut_ports[1], "fastleave": "on"}]}])
+    err_msg = f"Verify that port entities set to fastleave ON.\n{out}"
+    assert out[0][dev_name]["rc"] == 0, err_msg
+
+    # 2.Init interfaces and create 2 mcast Streams
+    # 3.Create 3 membership report streams, 1 stream with invalid checksum
+    dev_groups = tgen_utils_dev_groups_from_config(
+        [{"ixp": tg_ports[0], "ip": "102.0.0.3", "gw": "102.0.0.1", "plen": 24},
+         {"ixp": tg_ports[1], "ip": "100.0.0.3", "gw": "100.0.0.1", "plen": 24},
+         {"ixp": tg_ports[2], "ip": "101.0.0.3", "gw": "101.0.0.1", "plen": 24},
+         {"ixp": tg_ports[3], "ip": "99.0.0.3", "gw": "99.0.0.1", "plen": 24}])
+
+    await tgen_utils_traffic_generator_connect(tgen_dev, tg_ports, dut_ports, dev_groups)
+    tx_port = dev_groups[tg_ports[0]][0]["name"]
+    rx_ports = [dev_groups[tg_ports[1]][0]["name"],
+                dev_groups[tg_ports[2]][0]["name"],
+                dev_groups[tg_ports[3]][0]["name"]]
+
+    streams = {f"mcast_{idx + 1}": {
+            "type": "raw",
+            "protocol": "ip",
+            "ip_source": tx_port,
+            "ip_destination": rx_ports[1],
+            "srcMac": macs[0],
+            "dstMac": mcast_ip_to_mac(dst_ip),
+            "srcIp": "0.0.0.0",
+            "dstIp": dst_ip,
+            "frame_rate_type": "line_rate",
+            "rate": 40,
+        } for idx, dst_ip in enumerate([mcast_group1, mcast_group2])}
+
+    streams.update({f"member_report{idx + 1}": {
+            "type": "raw",
+            "protocol": "ip",
+            "ip_source": dev_groups[tg_ports[idx + 1]][0]["name"],
+            "ip_destination": dev_groups[tg_ports[0]][0]["name"],
+            "srcMac": combined[1],
+            "dstMac": mcast_ip_to_mac(combined[0]),
+            "srcIp": dev_groups[tg_ports[idx + 1]][0]["ip"],
+            "dstIp": combined[0],
+            "ipproto": igmpv_and_type[igmp_msg_ver]["igmp_msg_v"],
+            "totalLength": igmpv_and_type[igmp_msg_ver]["totalLength"],
+            "igmpType": igmpv_and_type[igmp_msg_ver]["igmp_type"],
+            "igmpGroupAddr": combined[0],
+            "rate": 1,
+            "transmissionControlType": "fixedPktCount",
+            "frameCount": 1
+        } for idx, combined in enumerate(zip(mcast_group_addr, macs[1:]))
+    })
+
+    streams["member_report3"]["igmpChecksum"] = "0"
+    if igmp_msg_ver == 3:
+        for i in range(1, 4):
+            streams[f"member_report{i}"]["numberOfSources"] = "1"

```

Comment:
I would advice not to use indexes in loops where possible
```suggestion
        for name, stream in streams.items():
            if "member_report" in name:
                stream["numberOfSources"] = "1"
```

Reply:
Updated